src/features/embalm/stepContent/hooks/useCreateRecipientPDF/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { useToast } from '@chakra-ui/react';
+import { useState, useCallback } from 'react';
+import { setRecipient, Recipient } from 'store/embalm/actions';
+import { useDispatch, useSelector } from 'store/index';
+import { Page, Text, Image, View, Document, StyleSheet, pdf } from '@react-pdf/renderer';
+import { saveAs } from 'file-saver';
+import QRCode from 'qrcode';
+import santiize from 'sanitize-filename';
+import { generateRecipientKeysAndPDF, generateRecipientPDFFailure } from 'lib/utils/toast';
+
+import { createEncryptionKeypairAsync } from '../useCreateEncryptionKeypair';
+import { ethers } from 'ethers';
+
+const document = async (name: string, recipient: Recipient) => {
+  const styles = StyleSheet.create({
+    page: {
+      flexDirection: 'column',
+      margin: 25,
+      fontSize: 11,
+    },
+    section: {
+      margin: 5,
+      padding: 0,
+      flexGrow: 1,
+    },
+    imageView: { height: 120, textAlign: 'center', width: '20%' },
+    image: { marginHorizontal: '10%', width: 'auto', height: 'auto' },
+  });
+
+  const qrCodeOption = { margin: 0 };
+
+  const addressQR = await QRCode.toDataURL(recipient.address, qrCodeOption);
+  const publicKeyQR = await QRCode.toDataURL(recipient.publicKey, qrCodeOption);
+  const privateKeyQR = await QRCode.toDataURL(recipient.privateKey || '', qrCodeOption);
+
+  return (
```

Comment:
Minor comment... It would be nice if the jsx that bulids the pdf were moved outside the hook. Don't let this hold up the PR though.